kani-compiler/src/kani_middle/transform/check_aliasing/cached_body_mutator.rs

```diff
@@ -0,0 +1,94 @@
+use std::collections::HashMap;
+use super::{BodyMutator, Local, MirInstance, Mutability, LocalDecl, Body, Ty, Statement, MutatorIndex, MutatorIndexStatus, Instruction, Span};
+
+/// Body mutator which wraps the BodyMutator
+/// interface with a cache of the locals that
+/// store function calls.
+pub struct CachedBodyMutator {
```

Comment:
It seems to me that the only functionality this struct introduces is caching function locals to use in instrumented calls and providing a unit local to use as a return target. I think those could be just incorporated into the transformation pass.